---
title: "Chap 2 - Hood vs. Bench"
author: "Linh Dang"
editor: visual
execute:
  warning: false
  message: false
format:
  html:
    code-fold: true
    toc: true
    toc-location: left
    toc-depth: 3
---

```{r message=FALSE}
library(phyloseq)
library(microViz)
library(microbiome)
library(vegan)
library(rstatix)
library(ggpubr)
library(Wrench)
library(patchwork)
library(lme4)       # For alpha diversity mixed models
library(lmerTest)
library(tidyverse)

rm(list = ls())
pseq <- rprojroot::find_rstudio_root_file() %>% 
  file.path("data/Chap2/px_J_U_NCT_v02.rds") %>% 
  readRDS() %>% 
  ps_get() %>% 
  tax_filter(min_total_abundance = 2, min_prevalence = 2)
## edit meta
pseq <- pseq %>% 
  ps_mutate(TA = ifelse(person == "J", "TA1", "TA2")) %>% 
  ps_mutate(Envir = HOOD_BENCH) %>% 
  ps_select(-c("HOOD_BENCH", "person"))
```

# Project Description

Investigate NCT samples in different environments: as normal (BENCH), and clean one (HOOD).

Common variables and functions

```{r echo=TRUE}
Kolors <- c("#9d547c","#56ca63","#a357d6","cornflowerblue","#419d2a","sandybrown","red3","peachpuff","cyan","paleturquoise3","mistyrose","mediumpurple","mediumseagreen","mediumorchid","moccasin","orange4","olivedrab","midnightblue","papayawhip","palevioletred4","brown1","greenyellow","orchid","navy","darkred","navajowhite1","mistyrose1","grey85","#525fd6","red2","#8cbe3a","#c944aa","indianred3","#5ba557","#9e66cb","#c1b735","#6d82ec","grey25","#e69728","#6654b0","lightsalmon3","lightcyan1","khaki1","seagreen1","plum1","lightsteelblue1","palevioletred3","mintcream","magenta3","#799330","#da7fdf","#3c782c","#e44586","blue4","#63c996","#dc3f53","#49cbc8","#cf3f29","#4fabda","#da6c2b","#598bd1","#b78c24","#8d4191","#a0b971","slategray1","sienna","plum1","lightyellow1","lightskyblue3","linen","limegreen","cornsilk1","mediumaquamarine","gray14","gold3","darkviolet","#b2386a","#479d71","#ae4341","#2ba198","#e07557","#5361a3","#dda353","#aa98df","#5b6114","#dc89bf","#327243","slateblue1","#e57b94","#277257","#9b62a0","#bbab59","#98495a","#526229","#d8827d","#857624","gray40","#9a4a22","#7c7d46","mediumslateblue","lemonchiffon1","#e3a073","#9e6b33", "gray74","slateblue1","rosybrown3", "lawngreen","gainsboro","dodgerblue3","deeppink3","firebrick3", "orchid2", "olivedrab1", "ivory3", "darkseagreen", "bisque2", "darkgoldenrod2", "blue2", "skyblue", "seashell2", "turquoise", "tan1", "seagreen2", "palevioletred3", "linen", "steelblue4","ghostwhite","dodgerblue1","deeppink1","firebrick1", "limegreen", "purple3", "khaki3", "snow3", "darkslategray","darkorchid","lavender", "magenta2", "palegreen", "salmon", "maroon", "cyan2","#671408","#FAEBD7","#7FFFD4","#F0FFFF","#A52A2A","burlywood","cadetblue","#7FFF00","chocolate","cornsilk","slateblue1","#FF7F50","red1","#008B8B","darkgoldenrod1","darkolivegreen","darkorange4","white","hotpink","honeydew1","goldenrod2","darkgreen","oldlace","darkslategray3","navajowhite3","orchid4","gray25","#F0924D")

## Alpha analysis
AlphaPlot_Violin_LMM <- function(ps, main_factor, batch_factor, metric = "Obsered"){
  ## ps: phyloseq object
  ## metric: Obsered, shannon, simpson, invsimpson
  ## df: in pivot longer manner
  ## calc p value in manner of paired
  ## get y.position for ggplot
  lst_grp <- meta(pseq)[[main_factor]] %>% unique() %>% sort()
  if(tolower(metric) == "obsered"){
    div <- ps %>% 
      otu_table() %>% 
      specnumber(MARGIN = 2)
    
  }else{
    div <- vegan::diversity(otu_table(ps), index = metric, MARGIN = 2)
  }
  df <- tibble(ID = names(div), Val = unname(div))
  df_meta <- meta(ps)
  df <- merge(df, df_meta, by = "ID")
  
  y_pos <- t_test(formula = as.formula(sprintf("Val ~ %s", main_factor)), data = df) %>% add_xy_position() %>% pull(y.position)
  if(!is.null(batch_factor)){
    model <- lmer(as.formula(sprintf("Val ~ %s + (1|%s)", main_factor, batch_factor)), data = df)
  }else{
    model <- lm(as.formula(sprintf("Val ~ %s", main_factor)), data = df)
  }
  
  #emm <- emmeans(model, specs = as.formula(sprintf("pairwise ~ %s", main_factor)))
  #contr <- contrast(emm, method = "pairwise")
  
  
  p_add <- model %>%
    anova() %>% 
    as_tibble() %>%
    transmute(
      group1 = lst_grp[1],
      group2 = lst_grp[2],
      p = `Pr(>F)`
      ##method = "emmeans model-based t-test"
    ) %>% 
    add_significance() %>% 
    mutate(y.position = y_pos)
  plt <- ggplot(df, aes(x = .data[[main_factor]], y = .data[["Val"]], fill = .data[[main_factor]])) +
    geom_violin() +
    #geom_boxplot(alpha = 0.6, outlier.shape = NA) +
    geom_boxplot(width=0.2) +
    geom_point(position = position_jitter(width = 0.1), alpha = 0.5) +
    # Connect lines for the same sample to show paired changes
    #geom_line(aes(group = AN_NR), color = "gray", alpha = 0.5) +
    theme_minimal() +
    labs(
      #title = "Alpha Diversity",
      title = sprintf("Alpha Diversity: %s", metric),
      y = metric
    ) +
    theme(legend.position = "none")
  plt <- plt + stat_pvalue_manual(p_add, label = "p", inherit.aes = FALSE, tip.length = 0.01)
  return(plt)
}
### Beta

## Wrench Normalization
WrenchWrapper <- function(PhyloObjct, grp, roundUp = F){
  cnt_table <- PhyloObjct %>% otu_table()
  group <- PhyloObjct %>% sample_data() %>% pull(grp)
  w <- wrench(cnt_table, condition = group)
  
  # deseq.obj <- DESeqDataSetFromMatrix(cnt_table %>% as.data.frame(), DataFrame(group), ~group)
  # DESeq2::sizeFactors(deseq.obj) <- w$nf
  # cnt_table_normalized <- DESeq2::counts(deseq.obj, normalized=TRUE)
  
  norm_factors <- w$nf
  norm_counts <- sweep(cnt_table, 2, norm_factors, FUN = '/')
  if(roundUp){norm_counts <- norm_counts %>% round()}
  if(!is.null(phy_tree(PhyloObjct, errorIfNULL = F))){
    return(phyloseq(otu_table(norm_counts, taxa_are_rows = T), tax_table(PhyloObjct %>% tax_table()), sample_data(PhyloObjct %>% sample_data()),
                    phy_tree(PhyloObjct)))
  } else{
    return(phyloseq(otu_table(norm_counts, taxa_are_rows = T), tax_table(PhyloObjct %>% tax_table()), sample_data(PhyloObjct %>% sample_data())))
  }
}
```

# Basic Analysis

## Read Count Overview

We look into the read count after trivial filtering of NCT samples.

```{r}
metadata <- meta(pseq)
df_readcount <- sample_sums(pseq) %>% enframe(name = "ID", value = "Reads")
df_readcount <- df_readcount %>% 
  left_join(., metadata, by = "ID")

ggplot(df_readcount, aes(x = TA, y = Reads, fill = Envir)) +
  geom_boxplot(width=0.2) +
    geom_point(position = position_jitter(width = 0.1), alpha = 0.5) +
  ##geom_col(position = "dodge") +
  labs(
    x = "Technician",
    y = "Read counts",
    fill = "Condition"
  ) +
  theme_minimal()
```

## Alpha Diversity

Plot Alpha diversity with p-value calculated via Linear Mixed Model (LMM) with fixed effect is condition (HOOD vs. BENCH) conditioned on random effect technicians.

Because the sequencing depth of each sample are not significantly various, I applied the rarefaction as normalization.

```{r}
pseq_rar <- rarefy_even_depth(pseq)
metadata <- microbiome::meta(pseq_rar)
alpha_div <- estimate_richness(pseq_rar, measures = c("Observed", "Shannon")) %>% 
  rownames_to_column(var = "ID")
df_alpha <- merge(metadata, alpha_div, by = "ID")
```

### Between Environments

Observed species w.r.t conditions, accounting for effect of technicians

```{r}
model_obs <- lmer(Observed ~ Envir + (1 | TA), data = df_alpha)
anova(model_obs) %>% print()
```

Shannon index w.r.t conditions, accounting for effect of technicians

```{r}
model_shannon <- lmer(Shannon ~ Envir + (1 | TA), data = df_alpha)
anova(model_shannon) %>% print()
```

Visualization

```{r}
plt1 <- AlphaPlot_Violin_LMM(pseq_rar, main_factor = "Envir", batch_factor = "TA", metric = "obsered")
plt2 <- AlphaPlot_Violin_LMM(pseq_rar, main_factor = "Envir", batch_factor = "TA", metric = "shannon")
plt1 + plt2 + plot_layout(guides = "collect")
```

### Between Technicians

As expected, the Alpha diversity between technicians are significantly different w.r.t richness, but not to number of observed species, as follow:

```{r}
plt1 <- AlphaPlot_Violin_LMM(pseq_rar, main_factor = "TA", batch_factor = NULL, metric = "obsered")
plt2 <- AlphaPlot_Violin_LMM(pseq_rar, main_factor = "TA", batch_factor = NULL, metric = "shannon")
plt1 + plt2 + plot_layout(guides = "collect")
```

## Beta Diversity

We investigate the beta diversity of samples w.r.t conditions (HOOD vs. BENCH) accounting for technicians.

### Analysis with permutation approach

We test with different transformation and different distance metric.

#### Transformation: CLR, Euclidean Space

```{r}
pseq_clr <- pseq %>% microbiome::transform(transform = "clr")
dist_matrix <- phyloseq::distance(pseq_clr, method = "euclidean")
perm <- how(blocks = metadata$TA, nperm = 99999)
adonis_res <- adonis2(dist_matrix ~ Envir, 
                      data = metadata, 
                      #permutations = 9999
                      permutations = perm
                      )

print(adonis_res)
```

#### Transformation: compositional, bray Space

```{r}
pseq_comp <- pseq %>% microbiome::transform(transform = "compositional")
dist_matrix <- phyloseq::distance(pseq_comp, method = "bray")
perm <- how(blocks = metadata$TA, nperm = 99999)
adonis_res <- adonis2(dist_matrix ~ Envir, 
                      data = metadata, 
                      #permutations = 9999
                      permutations = perm
                      )

print(adonis_res)
```

For the comparison, here is the adoinis2 without accounting effect of technicians

```{r}
pseq_comp <- pseq %>% microbiome::transform(transform = "compositional")
dist_matrix <- phyloseq::distance(pseq_comp, method = "bray")
#perm <- how(blocks = metadata$TA, nperm = 99999)
adonis_res_free <- adonis2(dist_matrix ~ Envir, 
                      data = metadata, 
                      permutations = 9999
                      #permutations = perm
                      )

print(adonis_res_free)
```

### Visualization

```{r}
dist_mx <- "euclidean"
otu_tab <- microbiome::abundances(pseq_clr)
# Function to get residuals from Technician effect
get_residuals <- function(taxa_vector, batch_factor) {
  # We fit: Abundance ~ Technician
  fit <- lm(taxa_vector ~ batch_factor)
  # We return the residuals
  return(residuals(fit))
}

otu_residuals <- apply(otu_tab, 1, function(x) get_residuals(x, metadata$TA))
otu_residuals <- t(otu_residuals)

ps_resid <- pseq_clr ##pseq_comp
otu_table(ps_resid) <- otu_table(otu_residuals, taxa_are_rows = TRUE)

# annotation
  annotations_before <- data.frame(
    xpos = c(-Inf),
    ypos =  c(Inf),
    annotateText = paste0("p-val=",adonis_res_free$`Pr(>F)`[[1]]),
    hjustvar = c(-0.2) ,
    vjustvar = c(1.5))
  annotations_after <- data.frame(
    xpos = c(-Inf),
    ypos =  c(Inf),
    annotateText = paste0("p-val=", adonis_res$`Pr(>F)`[[1]]),
    hjustvar = c(-0.2) ,
    vjustvar = c(1.5))

# Before (Original CLR)
m_group <- "Envir"
ord_before <- ordinate(pseq_comp, method = "PCoA", distance = dist_mx)
p1 <- plot_ordination(pseq_comp, ord_before, color = "Envir", shape = "TA") + ## 
  stat_ellipse(aes(group = .data[[m_group]]), linewidth = 0.3) +
  theme_bw() + ggtitle("Before") +
  geom_text(data=annotations_before,aes(x=xpos,y=ypos,hjust=hjustvar,vjust=vjustvar,
                                   label=annotateText), size = 4.5, inherit.aes = FALSE)

# After (Residuals)
ord_after <- ordinate(ps_resid, method = "PCoA", distance = dist_mx)
p2 <- plot_ordination(ps_resid, ord_after, color = "Envir", shape = "TA") +
  stat_ellipse(aes(group = .data[[m_group]]), linewidth = 0.3) +
  theme_bw() + ggtitle("After") +
  geom_text(data=annotations_after,aes(x=xpos,y=ypos,hjust=hjustvar,vjust=vjustvar,
                                   label=annotateText), size = 4.5, inherit.aes = FALSE)
grid.newpage()
p1 + p2 + plot_layout(guides = "collect")
```

For slide

```{r}
##| fig-width: 16
##| fig-height: 9
##| fig-lightbox: true
m_group <- "Envir"
ord_before <- ordinate(pseq_comp, method = "PCoA", distance = dist_mx)
p1 <- plot_ordination(pseq_comp, ord_before, color = "Envir", shape = "TA") + ## 
  stat_ellipse(aes(group = .data[["Envir"]])) +
  theme_bw() + ggtitle("Grouped by Environment") +
  geom_text(data=annotations_before,aes(x=xpos,y=ypos,hjust=hjustvar,vjust=vjustvar,
                                   label=annotateText), size = 4.5, inherit.aes = FALSE)

# After (Residuals)
ord_after <- ordinate(ps_resid, method = "PCoA", distance = dist_mx)
p2 <- plot_ordination(ps_resid, ord_before, color = "Envir", shape = "TA") +
  stat_ellipse(aes(group = .data[["TA"]])) +
  theme_bw() + ggtitle("Grouped by Technician") #+
  #geom_text(data=annotations_after,aes(x=xpos,y=ypos,hjust=hjustvar,vjust=vjustvar, label=annotateText), size = 4.5, inherit.aes = FALSE)
grid.newpage()
p1 + p2 + plot_layout(guides = "collect")
```

**Remarks**: With compositional transformation and bray distance, the NCT samples are significantly different between HOOD vs. BENCH when accounting for technicians.

**CLR transformation with Limma**

```{r}
require(limma)
# 1. Transform to CLR (Centered Log-Ratio)
ps_clr <- microbiome::transform(pseq, "clr")
metadata <- data.frame(sample_data(ps_clr))

# 2. Extract OTU table
otu_tab <- as(otu_table(ps_clr), "matrix")

# 3. Create "Batch-Corrected" OTU table
# We remove 'Technician' but PROTECT 'Environment'
design <- model.matrix(~Envir, data = metadata)
otu_corrected <- removeBatchEffect(otu_tab, batch = metadata$TA, design = design)

# 4. Create a corrected phyloseq object for plotting
ps_corrected <- ps_clr
otu_table(ps_corrected) <- otu_table(otu_corrected, taxa_are_rows = TRUE)
```

Before and after for visualization

```{r}
# A. Before Correction: PCoA showing Technician dominance
ord_before <- ordinate(ps_clr, method = "PCoA", distance = "euclidean") # euclidean
p1 <- plot_ordination(ps_clr, ord_before, color = "Envir", shape = "TA") +
  geom_point(size = 3, alpha = 0.7) +
  stat_ellipse(aes(group = Envir)) +
  theme_bw() +
  ggtitle("A. Raw Data (Unsignificant)", 
          subtitle = "Technician variance masks the Environment signal")

# B. After Correction: PCoA showing the 'Hidden' Environment signal
ord_after <- ordinate(ps_corrected, method = "PCoA", distance = "euclidean")
p2 <- plot_ordination(ps_corrected, ord_after, color = "Envir", shape = "TA") +
  geom_point(size = 3, alpha = 0.7) +
  stat_ellipse(aes(group = Envir)) +
  theme_bw() +
  ggtitle("B. Corrected Data (Significant)", 
          subtitle = "Technician variance removed; Environment signal revealed")

# Combine side-by-side
grid.newpage()
p1 + p2 + plot_layout(guides = "collect")
```

Alternative

```{r}
# We use the raw CLR ordination but connect samples by Technician
p3 <- plot_ordination(ps_clr, ord_before, color = "Envir") +
  geom_point(aes(shape = TA), size = 3) +
  # This line connects the samples handled by the same technician
  geom_line(aes(group = TA), arrow = arrow(length=unit(0.2,"cm")), alpha = 0.4) +
  theme_minimal() +
  ggtitle("C. Directional Shift Plot", 
          subtitle = "Arrows show Environment change within each Technician batch")

p3
```

# DAA

Here we investigate the DDA species between environment after accounting for the technician. Due to the small number of samples, Maaslin-based approaches are not appropriate. We would use the ANCOM-BC and ALDEx2 instead.

## Between Environments

### By ANCOM-BC

```{r}
#require(Maaslin2
# require("ANCOMBC")
# 
# res <- ancombc2(
#   data = pseq %>% tax_filter(min_prevalence = 0.15, min_total_abundance = 1e-4),
#   rank = "genus",
#   fix_formula = "Envir",
#   rand_formula = "(1|TA)",
#   group = NULL,             # Variable for multi-group comparison/structural zeros
#   struc_zero = FALSE,               # Recommended to detect absolute presence/absence
#   neg_lb = TRUE,                   # Better structural zero detection for larger samples
#   alpha = 0.1,                    # Significance level
#   n_cl = 2 
# )
# saveRDS(res, "./ancombc2.rds")
```

Visualization

```{r}
# Filter for significant results
res <- readRDS("./ancombc2.rds")
res_prim <- res$res
sig_taxa <- res_prim %>%
  filter(p_EnvirHOOD <=0.1)
  #filter(diff_ConditionName == TRUE) %>% # Replace ConditionName with your actual group
  #arrange(desc(abs(lfc_ConditionName))) %>%
  #head(20)

# Plotting
ggplot(sig_taxa, aes(x = reorder(taxon, lfc_EnvirHOOD), y = lfc_EnvirHOOD, fill = lfc_EnvirHOOD > 0)) +
  geom_bar(stat = "identity") +
  coord_flip() +
  labs(title = "Differential Abundance (Accounting for Technician)",
       x = "Taxon", y = "Log Fold Change") +
  theme_minimal()
```

### By ALDEx2

```{r}
require(ALDEx2)
otu_count <- microbiome::abundances(pseq)
metadata <- microbiome::meta(pseq)
mm <- model.matrix(~ Envir + TA, data = metadata)
# Step A: Generate Monte Carlo samples of the Dirichlet distribution
# This accounts for the uncertainty of small sample sizes
x <- aldex.clr(otu_count, mm, mc.samples = 128, denom = "all", verbose = FALSE)
# Step B: Run the GLM
# This will test every taxon against the Condition AND Technician
glm_result <- aldex.glm(x, mm)
# Step C: Calculate Effect Sizes (Crucial for ALDEx2 plotting)
# This calculates the 'difference' specifically for the Condition
glm_effect <- aldex.glm.effect(x)
```

```{r}
res_tidy <- glm_result %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "Taxon") %>% 
  dplyr::select(Taxon,
         p_val = `EnvirHOOD:pval`,
         q_val = `EnvirHOOD:pval.padj`,
         estimate = `EnvirHOOD:Est`) %>% 
  mutate(is_sig = p_val <= 0.2)
```

```{r}
aldex.glm.plot(glm_result, eff = glm_effect, contrast = "EnvirHOOD", type = "MW")
```

```{r}
top_hits <- res_tidy %>%
  filter(is_sig) %>%
  arrange(p_val) %>%
  head(20)

ggplot(top_hits, aes(x = reorder(Taxon, estimate), y = estimate, fill = estimate > 0)) +
  geom_bar(stat = "identity") +
  coord_flip() +
  scale_fill_manual(values = c("skyblue", "salmon"), labels = c("Down", "Up"), name = "Direction") +
  labs(title = "Top Significant Taxa (ALDEx2)",
       subtitle = "Adjusted for Technician Batch Effect",
       x = "Taxon", y = "GLM Coefficient (Estimate)") +
  theme_minimal()
```

Remarks: Sphingomonas genus tend to be more abundance in normal environment (on the bench), while Variovorax genus tend to dominant in clean environment (under the hood)
