% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
\documentclass[
]{article}
\usepackage{xcolor}
\usepackage[margin=1in]{geometry}
\usepackage{amsmath,amssymb}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math} % this also loads fontspec
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else
  % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\newsavebox\pandoc@box
\newcommand*\pandocbounded[1]{% scales image to fit in text height/width
  \sbox\pandoc@box{#1}%
  \Gscale@div\@tempa{\textheight}{\dimexpr\ht\pandoc@box+\dp\pandoc@box\relax}%
  \Gscale@div\@tempb{\linewidth}{\wd\pandoc@box}%
  \ifdim\@tempb\p@<\@tempa\p@\let\@tempa\@tempb\fi% select the smaller of both
  \ifdim\@tempa\p@<\p@\scalebox{\@tempa}{\usebox\pandoc@box}%
  \else\usebox{\pandoc@box}%
  \fi%
}
% Set default figure placement to htbp
\def\fps@figure{htbp}
\makeatother
% definitions for citeproc citations
\NewDocumentCommand\citeproctext{}{}
\NewDocumentCommand\citeproc{mm}{%
  \begingroup\def\citeproctext{#2}\cite{#1}\endgroup}
\makeatletter
 % allow citations to break across lines
 \let\@cite@ofmt\@firstofone
 % avoid brackets around text for \cite:
 \def\@biblabel#1{}
 \def\@cite#1#2{{#1\if@tempswa , #2\fi}}
\makeatother
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newenvironment{CSLReferences}[2] % #1 hanging-indent, #2 entry-spacing
 {\begin{list}{}{%
  \setlength{\itemindent}{0pt}
  \setlength{\leftmargin}{0pt}
  \setlength{\parsep}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1
   \setlength{\leftmargin}{\cslhangindent}
   \setlength{\itemindent}{-1\cslhangindent}
  \fi
  % set entry spacing
  \setlength{\itemsep}{#2\baselineskip}}}
 {\end{list}}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{\hfill\break\parbox[t]{\linewidth}{\strut\ignorespaces#1\strut}}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{\strut#1\strut}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{\strut#1\strut}}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\usepackage{bookmark}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same}
\hypersetup{
  pdftitle={My DPC Manuscript},
  pdfauthor={Linh Dang},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{My DPC Manuscript}
\author{Linh Dang}
\date{}

\begin{document}
\maketitle

\section{Manusctipt Title}\label{manusctipt-title}

\subsection{Abstract}\label{abstract}

\subsection{Introduction}\label{introduction}

\subsection{Results}\label{results}

\subsubsection{Lab Contamination Survey}\label{lab-contamination-survey}

\paragraph{Sample Statistcs}\label{sample-statistcs}

To address the issue of contamination, we comprehensively collected a
large number of negative control samples (NCT) at each stage of the
wet-lab sample processing workflow (Fig. xxx), including \textbf{93}
paraffin controls, \textbf{133} buffer controls, \textbf{11} PCR
controls, and \textbf{3} sequencing controls, resulting in a total of
6128 taxa. We remove sequencing control samples due to insufficient
reads (≤ 750) as well as taxa with extremely low prevalence and
abundance. With that criteria, we further filter out and result in
\textbf{203} negative control samples, comprising \textbf{113} buffer
controls, \textbf{84} paraffin controls, and \textbf{6} PCR control.
That results in \textbf{1775} taxa after rarefaction as normalization.
These negative controls were accumulated over several years, with a
significant expansion in control type paraffin and buffer introduced in
2022 (Table 1 ).

\textbf{Table 1.} Negative control samples collected over years.

\begin{longtable}[]{@{}lllll@{}}
\toprule\noalign{}
Sample ↓ / Feature → & 2021 & 2022 & 2023 & 2024 \\
\midrule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
Buffer & 29 & 60 & 24 & 0 \\
Paraffin & 5 & 57 & 11 & 12 \\
PCR & 0 & 3 & 3 & 0 \\
\end{longtable}

\paragraph{Alpha \& Beta Diversity}\label{alpha-beta-diversity}

Significant differences in microbial profiles (alpha and beta diversity)
were observed across different control types (Fig. 1 \& 2). Similar
patterns were also found for other factors such as technician, year, and
season of sequencing (in Supplementary Materials Figure xxx). In buffer
samples, the number of species and their evenness statistically
significant than the others.

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\linewidth - 4\tabcolsep) * \real{0.3151}}
  >{\raggedright\arraybackslash}p{(\linewidth - 4\tabcolsep) * \real{0.3151}}
  >{\raggedright\arraybackslash}p{(\linewidth - 4\tabcolsep) * \real{0.3699}}@{}}
\toprule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Alpha_sampletype_obseredSpecies.png}}
&
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Alpha_sampletype_Shannon.png}}
&
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Alpha_sampletype_InvSimpson.png}} \\
\end{longtable}

\textbf{Figure 1.} (A--C) Alpha diversity of NCT types w.r.t number of
observed species(A), Shannon Index(B), and inversed Simpson index(C).

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\linewidth - 2\tabcolsep) * \real{0.5000}}
  >{\raggedright\arraybackslash}p{(\linewidth - 2\tabcolsep) * \real{0.5000}}@{}}
\toprule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Beta_rar_sampleType.png}}
&
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Beta_Wrench_SampleType.png}} \\
\end{longtable}

\textbf{Figure 2.} (A-B) Beta diversity of various NCT samples, with
rarefaction and Wrench normalization respectively.

\paragraph{Microbial Profile of NCT
Samples}\label{microbial-profile-of-nct-samples}

The 20 most abundant taxa in whole set of NCT represented a mixture of
known environmental microbes and potential human commensals (Fig. 3).
For example, \emph{Sphingomonas}, a well-known environmental taxon
frequently found in hospital settings, was detected with high abundance
in nearly all negative controls. In contrast, human-associated taxa such
as \emph{Veillonella parvula}, previously reported in PDAC-related52
studies\textsuperscript{1}, were present in approximately 60 percent of
NCT samples but at significantly lower abundance (Fig.3). This
highlights the importance of not discarding all taxa found in negative
controls, but instead applying appropriate decontamination
approaches\textsuperscript{2--4} to systematically remove likely
environmental contaminants.

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\linewidth - 2\tabcolsep) * \real{0.5000}}
  >{\raggedright\arraybackslash}p{(\linewidth - 2\tabcolsep) * \real{0.5000}}@{}}
\toprule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Heatmap_wrench_species_bysum.png}}
&
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Heatmap_wrench_genus_bysum.png}} \\
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Barplot_SampleType_sum.png}}
&
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Barplot_SampleType_prev.png}} \\
\end{longtable}

\textbf{Figure 3.} Bacterial heatmap of NCT samples by species and genus
(top, left to right) of top 20 most abundance taxa. Bacterial barplot of
top 20 genus order by abundance and prevalence respectively (bottom,
left to right).

Beside negative control sample types, Contaminants in the laboratory
could be strongly affected by seasons. \ldots\ldots{}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\linewidth - 2\tabcolsep) * \real{0.5000}}
  >{\raggedright\arraybackslash}p{(\linewidth - 2\tabcolsep) * \real{0.5000}}@{}}
\toprule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Taxatree_sampletype_season.png}}
&
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Taxatree_sampletype_season_key.png}} \\
\end{longtable}

\textbf{Figure 4.}

\subsubsection{Contamination - Hood versus
Bench}\label{contamination---hood-versus-bench}

We were intrigued by how human commensals might enter negative control
samples. To investigate this, we conducted an experiment in which
negative samples were processed by two technicians under different
environmental conditions, as described in Fig. 2A and detailed in the
Materials and Methods section. Analysis of alpha diversity revealed no
significant differences across environmental conditions, instead the
person has an opposite direction. (Fig. 2B). Similarly, beta diversity
analysis showed that samples clustered significantly by technician
(\hlred{$p < 0.001$}), but not by environmental conditions alone
(\hlred{$p = 0.596$}; Fig. 2C).

\begin{longtable}[]{@{}ll@{}}
\toprule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap2/Alpha_Cond.png}}
&
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap2/Alpha_TA.png}} \\
\end{longtable}

\textbf{Figure 5.} Alpha diversity of NCT sample w.r.t environmental
conditions(left) and technicians (right).

\pandocbounded{\includegraphics[keepaspectratio]{img/Chap2/Beta_wrench_bray.png}}

\textbf{Figure 6.} Beta diversity cluster by conditions and technicians
respectively.

\subsubsection{Intratumor Bacterial Profile of Fresh Frozen
Samples}\label{intratumor-bacterial-profile-of-fresh-frozen-samples}

\subsection{Discussion}\label{discussion}

\subsection*{References}\label{references}
\addcontentsline{toc}{subsection}{References}

\phantomsection\label{refs}
\begin{CSLReferences}{0}{0}
\bibitem[\citeproctext]{ref-McKinley2023}
\CSLLeftMargin{1. }%
\CSLRightInline{McKinley, K. N. L. \emph{et al.}
\href{https://doi.org/10.3390/microorganisms11061466}{Translocation of
oral microbiota into the pancreatic ductal adenocarcinoma tumor
microenvironment}. \emph{Microorganisms} \textbf{11}, 1466 (2023).}

\bibitem[\citeproctext]{ref-Austin2023}
\CSLLeftMargin{2. }%
\CSLRightInline{Austin, G. I. \emph{et al.}
\href{https://doi.org/10.1038/s41587-023-01696-w}{Contamination source
modeling with SCRuB improves cancer phenotype prediction from microbiome
data}. \emph{Nature Biotechnology} \textbf{41}, 1820--1828 (2023).}

\bibitem[\citeproctext]{ref-Davis2018}
\CSLLeftMargin{3. }%
\CSLRightInline{Davis, N. M., Proctor, D. M., Holmes, S. P., Relman, D.
A. \& Callahan, B. J.
\href{https://doi.org/10.1186/s40168-018-0605-2}{Simple statistical
identification and removal of contaminant sequences in marker-gene and
metagenomics data}. \emph{Microbiome} \textbf{6}, (2018).}

\bibitem[\citeproctext]{ref-Nejman2020}
\CSLLeftMargin{4. }%
\CSLRightInline{Nejman, D. \emph{et al.}
\href{https://doi.org/10.1126/science.aay9189}{The human tumor
microbiome is composed of tumor type--specific intracellular bacteria}.
\emph{Science} \textbf{368}, 973--980 (2020).}

\end{CSLReferences}

\end{document}
