---
title: "Contamination Study"
subtitle: "Investigate Lab Intrinsic Contamination and PDAC intratumor microbiome"
author: "Linh Dang"
format:
  revealjs: 
    theme:  sky
    transition: fade
    fig-responsive: true
    width: 1250
    height: 700
    margin: 0.1
    min-scale: 0.2  # Allows the content to shrink significantly
    max-scale: 2.0  # Allows content to grow
    ##slide-number: true
    chalkboard: true # Interactive drawing in HTML
    controls-layout: bottom-right
    slide-number: true
    center: true
    auto-stretch: true
    menu:                      # Adds a table of contents sidebar
      side: left
      width: wide
  beamer:
    aspectratio: 169
    fig-pos: 'H'
    theme: Madrid
---

## Introduction to the Project

This presentation covers the following sub-projects.

-   **Contamination Survey**: Collection of negative controls (NCT) in the lab over period 2021-2024.
-   **NCT Profile**: Investigate bacterial profile of NCT samples under various environments.
-   **Microbiome in intratumor PDAC**: Benchmarking various decontamination approaches for investigation of microbiome in PDAC
-   **Suggested Pipeline**:

------------------------------------------------------------------------

## Contamination Survey

::::::: columns
:::: {.column width="40%"}
::: r-fit-text
***Original NCT collection***:

-   **93** paraffin controls,
-   **133** buffer controls,
-   **11** PCR controls,
-   **3** sequencing controls.

Resulting in **6128** taxa.
:::
::::

:::: {.column width="60%"}
::: r-fit-text
***Trivial Filtering***;

-   Remove NCT samples with insufficient reads (≤ 750).
-   Taxa filtering with following criteria:
    -   threshold for prevalence detection: 2 reads
    -   Minimum prevalence: 2 samples
    -   Minimum abundance: 1e-6

Resulting in **1775** taxa, consist of:

-   **84** paraffin controls,
-   **133** buffer controls,
-   **6** PCR controls.
:::
::::
:::::::

------------------------------------------------------------------------

## Diversity Analyses

**Alpha Diversity w.r.t Sample Types**

:::::: columns
::: {.column width="30%"}
![](img/Chap1/Alpha_sampletype_obseredSpecies.png)
:::

::: {.column width="30%"}
![](img/Chap1/Alpha_sampletype_Shannon.png)
:::

::: {.column width="30%"}
![](img/Chap1/Alpha_sampletype_InvSimpson.png)
:::
::::::

------------------------------------------------------------------------

## Beta diversity

**Beta diversity with rarefaction and Wrench as normalization**

::::: columns
::: {.column width="45%"}
![](img/Chap1/Beta_rar_sampleType.png)
:::

::: {.column width="45%"}
![](img/Chap1/Beta_Wrench_SampleType.png)
:::
:::::

------------------------------------------------------------------------

## Bacterial Profile - Barplot

::::: columns
::: {.column width="47%"}
**Top 15 by abundance** ![](img/Chap1/Barplot_SampleType_sum.png)
:::

::: {.column width="47%"}
**Top 15 by prevalence** ![](img/Chap1/Barplot_SampleType_prev.png)
:::
:::::

------------------------------------------------------------------------

## Bacterial Profile - Heatmap

::::: columns
::: {.column width="48%"}
**Top 20 species order by abundance** ![](img/Chap1/Heatmap_wrench_species_bysum.png)
:::

::: {.column width="48%"}
**Top 20 genus order by abundance** ![](img/Chap1/Heatmap_wrench_genus_bysum.png)
:::
:::::

------------------------------------------------------------------------

## Trend in Contamination

We applied DAA to find trends in contamination.

::::: columns
::: {.column width="48%"}
![](img/Chap1/Taxatree_sampletype_season.png)
:::

::: {.column width="48%"}
![](img/Chap1/Taxatree_sampletype_season_key.png)
:::
:::::

------------------------------------------------------------------------

## Hood vs. Bench

**Experiment Setup**

------------------------------------------------------------------------

## Hood vs. Bench - Alpha Diversity

::::: columns
::: {.column width="48%"}
Between technicians ![](img/Chap2/Chap2_Alpha_TA.png)
:::

::: {.column width="48%"}
Between environments after accounting for technicians ![](img/Chap2/Chap2_Alpha_Envir.png)
:::
:::::

------------------------------------------------------------------------

## Hood vs. Bench - Beta Diversity

NCT samples before and after accounting for the effect of technician by constrains permutation.

![](img/Chap2/Chap2_Beta_before_after.png)

------------------------------------------------------------------------

## Hood vs. Bench - DAA

::::: columns
::: {.column width="48%"}
By ANCOM-BC2 with p-val 0.1 ![](img/Chap2/Chap2_DAA_Envir_ANCOMBC.png)
:::

::: {.column width="48%"}
By ALDEx2 with p-val 0.2 ![](img/Chap2/Chap2_DAA_Envir_ALDEx2.png)
:::
:::::

------------------------------------------------------------------------

## PDAC intratumor microbiome

::::: columns
::: {.column width="40%"}
**Approaches**:

-   Raw (baseline)
-   Restrictive
-   decontam (R package)
-   SCRuB
-   Nejman approach.
:::

::: {.column width="60%"}
**Metrics for evaluation**:

-   Assessment of batch effect removal,
-   Divergence from Negative Controls,
-   Biologically meaningful of Consensus taxa,
-   Overlap with known true taxa / contaminants
:::
:::::

------------------------------------------------------------------------

## Are decontamination approaches different?

Alpha Diversities

:::::: columns
::: {.column width="33%"}
![](img/Chap3/Chap3_Alpha_Obs.png)
:::

::: {.column width="33%"}
![](img/Chap3/Chap3_Alpha_Shannon.png)
:::

::: {.column width="33%"}
![](img/Chap3/Chap3_Alpha_InvS.png)
:::
::::::

------------------------------------------------------------------------

## Beta Diversity

:::: {.columns}
::: {.column width="33%"}
![](img/Chap3/Chap3_Beta_Raw_decontam.png)
:::

::: {.column width="33%"}
![](img/Chap3/Chap3_Beta_Raw_restrictuve.png)
:::

::: {.column width="33%"}
![](img/Chap3/Chap3_Beta_Raw_SCRuB.png)
:::
::::

:::: columns
::: {.column width="33%"}
![](img/Chap3/Chap3_Beta_Raw_Nejman.png)
:::

::: {.column width="33%"}
![](img/Chap3/Chap3_Beta_all.png)
:::
::::

------------------------------------------------------------------------

## Bacterial Profile - Barplot

![](img/Chap3/Chap3_barplot.png)

------------------------------------------------------------------------

## Bacterial Profile - Heatmap

![](img/Chap3/Chap3_heatmap.png)

------------------------------------------------------------------------

## Overlap Taxa among Approaches - Order

![](img/Chap3/UpSet_order_WithoutNCT.png)

--------------------------------------------

## Overlap Taxa among Approaches - Class

![](img/Chap3/UpSet_class_woNCT.png)

--------------------------------------------

## Overlap Taxa among Approaches - Phylum

![](img/Chap3/UpSet_phylum_woNCT.png)

-------------------------------------------

## Overlap with Contaminants in NCT (prev 20%)

![](img/Chap3/UpSet_order_WithNCT.png)

---------------------------------------

## Quantify Assessment of Decontamination Approaches

-   **Assessment of batch effect removal**
-   <span style="color: gray;">Divergence from Negative Controls</span>
-   <span style="color: gray;">Biologically meaningful of Consensus taxa</span>
-   <span style="color: gray;">Overlap with known true taxa / contaminants</span>

----------------------------------------------

## Assessment of batch effect removal

-   ***Main idea***: A good decontamination method should reduce technical variation. In this case is the technician.
-   ***Approach***: Perform Ordination (PCoA) and PERMANOVA (adonis2).
-   ***Metric***: Check if the variance explained (R2) by “Sequencing Run” or “Extraction Batch” decreases after decontamination.
-   ***Logic***: Real tumor microbiomes shouldn’t cluster by which technician sequenced.

---------------------------------------------------

## Assessment of batch effect removal - Restrictive

![](img/Chap3/Chap3_Assess1_Restrictive.png)

--------------------------------------------

## Assessment of batch effect removal - Decontam

![](img/Chap3/Chap3_Assess1_Decontam.png)

---------------------------------------------
## Assessment of batch effect removal - SCRuB

![](img/Chap3/Chap3_Assess1_SCRuB.png)

---------------------------------------------

## Assessment of batch effect removal - Nejman

![](img/Chap3/Chap3_Assess1_Nj.png)

-------------------------------------

## Assessment of batch effect removal - Overall

:::: columns
::: {.column width="50%"}
|               | **R2** | **p-val**  |
|:--------------|:------:|-----------:|
| *Raw*         | 7.53%  | 0.1776     |
| *Restrictive* | 9.74%  | 0.0195     |
| *Decontam*    | 8.88%  | 0.0352     |
| *SCRuB*       | 9.2%   | 0.0291     |
| *Nejman*      | 9.17%  | 0.0486     |

:::

::: {.column width="50%"}
***Remarks***:

-   Before decontamination: samples are not significantly clustered w.r.t technicians/years.
-   After decontamination: samples are significantly clustered w.r.t technicians/years.

:::
::::

------------------------------------------

## Quantify Assessment of Decontamination Approaches

-   <span style="color: gray;">Assessment of batch effect removal</span>
-   **Divergence from Negative Controls**
-   <span style="color: gray;">Biologically meaningful of Consensus taxa</span>
-   <span style="color: gray;">Overlap with known true taxa / contaminants</span>

-----------------------------------------------

## Divergence from Negative Controls

-   ***Approach***: perform Inter-group dissimilarity and Wilcox paired test, following by visualization.
-   ***Metric***: Calculate the Jensen-Shannon Divergence (or Bray-Curtis) between samples and the pool of negative controls.
-   ***Logic***: the decontamination method should maximize the dissimilarity between true PDAC samples and NCTs.

-----------------------------------------------

## Divergence from Negative Controls - Plots

:::: columns
::: {.column width="50%"}
![](img/Chap3/Chap3_Assess2_restrictive.png){width=55%}
![](img/Chap3/Chap3_Assess2_decontam.png){width=55%}
:::
::: {.column width="50%"}
![](img/Chap3/Chap3_Assess2_SCRuB.png){width=55%}
![](img/Chap3/Chap3_Assess2_Nj.png){width=55%}
:::
::::

--------------------------------------------

## Remarks

-   Restrictive method: remove all taxa belongs to negative controls. Thus the taxa profile of decontaminated samples and negative controls are always different.
-   All methods seem not to be differentiated. 

-----------------------------------------------

## Quantify Assessment of Decontamination Approaches

-   <span style="color: gray;">Assessment of batch effect removal</span>
-   <span style="color: gray;">Divergence from Negative Controls</span>
-   <span style="color: gray;">Biologically meaningful of Consensus taxa</span>
-   **Overlap with known true taxa / contaminants**

-----------------------------------------------------

## Overlap with known true taxa / contaminants

::: {.r-fit-text}
-   **Approach**: design a measure quantifying the quality of decontamination approach by balancing the yield and purity explained below.
-   **Metric**: 
    -   *Yield* &rarr; number of putative significant true taxa not in NCTs / number of observed species. 
    -   *Purity* &rarr; number of putative significant true taxa not in NCTs / whole number of putative contaminants.
-   **Logic**: a good decontamination methods should give results whose taxa is less overlap with known high prevalence contaminants in large set of NCTs, while not over-removing true taxa.

Putative significant true taxa: by prevalence, by abundance, combination.
:::

--------------------------------------------------

### Abundance-based Approach

:::: columns
::: {.column width="50%"}
Check normality assumption

![](img/Chap3/Chap3_Assess4_Abund_check.png)
:::
::: {.column width="50%"}
![](img/Chap3/Chap3_Assess4_Abund.png)
:::
::::

---------------------------------------------

### Prevalence-based Approach

:::: columns
::: {.column width="50%"}
Check normality assumption
![](img/Chap3/Chap3_Assess4_Prev_Checked.png)
:::
::: {.column width="50%"}
![](img/Chap3/Chap3_Assess4_Prev.png)
:::
::::

-------------------------------------------

### Combination of Abundance and Prevalence

:::: columns
::: {.column width="50%"}
Check normality assumption
![](img/Chap3/Chap3_Assess4_Comb_check.png)
:::
::: {.column width="50%"}
![](img/Chap3/Chap3_Assess4_Comb.png)
:::
::::