% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
  ignorenonframetext,
  aspectratio=169,
]{beamer}
\usepackage{pgfpages}
\setbeamertemplate{caption}[numbered]
\setbeamertemplate{caption label separator}{: }
\setbeamercolor{caption name}{fg=normal text.fg}
\beamertemplatenavigationsymbolsempty
% Prevent slide breaks in the middle of a paragraph
\widowpenalties 1 10000
\raggedbottom
\setbeamertemplate{part page}{
  \centering
  \begin{beamercolorbox}[sep=16pt,center]{part title}
    \usebeamerfont{part title}\insertpart\par
  \end{beamercolorbox}
}
\setbeamertemplate{section page}{
  \centering
  \begin{beamercolorbox}[sep=12pt,center]{section title}
    \usebeamerfont{section title}\insertsection\par
  \end{beamercolorbox}
}
\setbeamertemplate{subsection page}{
  \centering
  \begin{beamercolorbox}[sep=8pt,center]{subsection title}
    \usebeamerfont{subsection title}\insertsubsection\par
  \end{beamercolorbox}
}
\AtBeginPart{
  \frame{\partpage}
}
\AtBeginSection{
  \ifbibliography
  \else
    \frame{\sectionpage}
  \fi
}
\AtBeginSubsection{
  \frame{\subsectionpage}
}

\usepackage{amsmath,amssymb}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\usetheme[]{Madrid}
\ifPDFTeX\else  
    % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\newif\ifbibliography
\setlength{\emergencystretch}{3em} % prevent overfull lines
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering


\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
\usepackage{caption}
% Make caption package work with longtable
\makeatletter
\def\fnum@table{\tablename~\thetable}
\makeatother
\usepackage{graphicx}
\makeatletter
\newsavebox\pandoc@box
\newcommand*\pandocbounded[1]{% scales image to fit in text height/width
  \sbox\pandoc@box{#1}%
  \Gscale@div\@tempa{\textheight}{\dimexpr\ht\pandoc@box+\dp\pandoc@box\relax}%
  \Gscale@div\@tempb{\linewidth}{\wd\pandoc@box}%
  \ifdim\@tempb\p@<\@tempa\p@\let\@tempa\@tempb\fi% select the smaller of both
  \ifdim\@tempa\p@<\p@\scalebox{\@tempa}{\usebox\pandoc@box}%
  \else\usebox{\pandoc@box}%
  \fi%
}
% Set default figure placement to htbp
\def\fps@figure{htbp}
\makeatother

\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\AtBeginDocument{%
\ifdefined\contentsname
  \renewcommand*\contentsname{Table of contents}
\else
  \newcommand\contentsname{Table of contents}
\fi
\ifdefined\listfigurename
  \renewcommand*\listfigurename{List of Figures}
\else
  \newcommand\listfigurename{List of Figures}
\fi
\ifdefined\listtablename
  \renewcommand*\listtablename{List of Tables}
\else
  \newcommand\listtablename{List of Tables}
\fi
\ifdefined\figurename
  \renewcommand*\figurename{Figure}
\else
  \newcommand\figurename{Figure}
\fi
\ifdefined\tablename
  \renewcommand*\tablename{Table}
\else
  \newcommand\tablename{Table}
\fi
}
\@ifpackageloaded{float}{}{\usepackage{float}}
\floatstyle{ruled}
\@ifundefined{c@chapter}{\newfloat{codelisting}{h}{lop}}{\newfloat{codelisting}{h}{lop}[chapter]}
\floatname{codelisting}{Listing}
\newcommand*\listoflistings{\listof{codelisting}{List of Listings}}
\makeatother
\makeatletter
\makeatother
\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\@ifpackageloaded{subcaption}{}{\usepackage{subcaption}}
\makeatother

\usepackage{bookmark}

\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same} % disable monospaced font for URLs
\hypersetup{
  pdftitle={Contamination Study},
  pdfauthor={Linh Dang},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}


\title{Contamination Study}
\subtitle{Investigate Lab Intrinsic Contamination and PDAC intratumor
microbiome}
\author{Linh Dang}
\date{}

\begin{document}
\frame{\titlepage}


\begin{frame}{Introduction to the Project}
\phantomsection\label{introduction-to-the-project}
This presentation covers the following sub-projects.

\begin{itemize}
\tightlist
\item
  \textbf{Contamination Survey}: Collection of negative controls (NCT)
  in the lab over period 2021-2024.
\item
  \textbf{NCT Profile}: Investigate bacterial profile of NCT samples
  under various environments.
\item
  \textbf{Microbiome in intratumor PDAC}: Benchmarking various
  decontamination approaches for investigation of microbiome in PDAC
\item
  \textbf{Suggested Pipeline}:
\end{itemize}
\end{frame}

\begin{frame}{Contamination Survey}
\phantomsection\label{contamination-survey}
\begin{columns}[T]
\begin{column}{0.4\linewidth}
\textbf{\emph{Original NCT collection}}:

\begin{itemize}
\tightlist
\item
  \textbf{93} paraffin controls,
\item
  \textbf{133} buffer controls,
\item
  \textbf{11} PCR controls,
\item
  \textbf{3} sequencing controls.
\end{itemize}

Resulting in \textbf{6128} taxa.
\end{column}

\begin{column}{0.6\linewidth}
\textbf{\emph{Trivial Filtering}};

\begin{itemize}
\tightlist
\item
  Remove NCT samples with insufficient reads (â‰¤ 750).
\item
  Taxa filtering with following criteria:

  \begin{itemize}
  \tightlist
  \item
    threshold for prevalence detection: 2 reads
  \item
    Minimum prevalence: 2 samples
  \item
    Minimum abundance: 1e-6
  \end{itemize}
\end{itemize}

Resulting in \textbf{1775} taxa, consist of:

\begin{itemize}
\tightlist
\item
  \textbf{84} paraffin controls,
\item
  \textbf{133} buffer controls,
\item
  \textbf{6} PCR controls.
\end{itemize}
\end{column}
\end{columns}
\end{frame}

\begin{frame}{Diversity Analyses}
\phantomsection\label{diversity-analyses}
\textbf{Alpha Diversity w.r.t Sample Types}

\begin{columns}[T]
\begin{column}{0.3\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Alpha_sampletype_obseredSpecies.png}}
\end{column}

\begin{column}{0.3\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Alpha_sampletype_Shannon.png}}
\end{column}

\begin{column}{0.3\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Alpha_sampletype_InvSimpson.png}}
\end{column}
\end{columns}
\end{frame}

\begin{frame}{Beta diversity}
\phantomsection\label{beta-diversity}
\textbf{Beta diversity with rarefaction and Wrench as normalization}

\begin{columns}[T]
\begin{column}{0.45\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Beta_rar_sampleType.png}}
\end{column}

\begin{column}{0.45\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Beta_Wrench_SampleType.png}}
\end{column}
\end{columns}
\end{frame}

\begin{frame}{Bacterial Profile - Barplot}
\phantomsection\label{bacterial-profile---barplot}
\begin{columns}[T]
\begin{column}{0.47\linewidth}
\textbf{Top 15 by abundance}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Barplot_SampleType_sum.png}}
\end{column}

\begin{column}{0.47\linewidth}
\textbf{Top 15 by prevalence}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Barplot_SampleType_prev.png}}
\end{column}
\end{columns}
\end{frame}

\begin{frame}{Bacterial Profile - Heatmap}
\phantomsection\label{bacterial-profile---heatmap}
\begin{columns}[T]
\begin{column}{0.48\linewidth}
\textbf{Top 20 species order by abundance}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Heatmap_wrench_species_bysum.png}}
\end{column}

\begin{column}{0.48\linewidth}
\textbf{Top 20 genus order by abundance}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Heatmap_wrench_genus_bysum.png}}
\end{column}
\end{columns}
\end{frame}

\begin{frame}{Trend in Contamination}
\phantomsection\label{trend-in-contamination}
We applied DAA to find trends in contamination.

\begin{columns}[T]
\begin{column}{0.48\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Taxatree_sampletype_season.png}}
\end{column}

\begin{column}{0.48\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap1/Taxatree_sampletype_season_key.png}}
\end{column}
\end{columns}
\end{frame}

\begin{frame}{Hood vs.~Bench}
\phantomsection\label{hood-vs.-bench}
\textbf{Experiment Setup}
\end{frame}

\begin{frame}{Hood vs.~Bench - Alpha Diversity}
\phantomsection\label{hood-vs.-bench---alpha-diversity}
\begin{columns}[T]
\begin{column}{0.48\linewidth}
Between technicians
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap2/Chap2_Alpha_TA.png}}
\end{column}

\begin{column}{0.48\linewidth}
Between environments after accounting for technicians
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap2/Chap2_Alpha_Envir.png}}
\end{column}
\end{columns}
\end{frame}

\begin{frame}{Hood vs.~Bench - Beta Diversity}
\phantomsection\label{hood-vs.-bench---beta-diversity}
NCT samples before and after accounting for the effect of technician by
constrains permutation.

\pandocbounded{\includegraphics[keepaspectratio]{img/Chap2/Chap2_Beta_before_after.png}}
\end{frame}

\begin{frame}{Hood vs.~Bench - DAA}
\phantomsection\label{hood-vs.-bench---daa}
\begin{columns}[T]
\begin{column}{0.48\linewidth}
By ANCOM-BC2 with p-val 0.1
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap2/Chap2_DAA_Envir_ANCOMBC.png}}
\end{column}

\begin{column}{0.48\linewidth}
By ALDEx2 with p-val 0.2
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap2/Chap2_DAA_Envir_ALDEx2.png}}
\end{column}
\end{columns}
\end{frame}

\begin{frame}{PDAC intratumor microbiome}
\phantomsection\label{pdac-intratumor-microbiome}
\begin{columns}[T]
\begin{column}{0.4\linewidth}
\textbf{Approaches}:

\begin{itemize}
\tightlist
\item
  Raw (baseline)
\item
  Restrictive
\item
  decontam (R package)
\item
  SCRuB
\item
  Nejman approach.
\end{itemize}
\end{column}

\begin{column}{0.6\linewidth}
\textbf{Metrics for evaluation}:

\begin{itemize}
\tightlist
\item
  Assessment of batch effect removal,
\item
  Divergence from Negative Controls,
\item
  Biologically meaningful of Consensus taxa,
\item
  Overlap with known true taxa / contaminants
\end{itemize}
\end{column}
\end{columns}
\end{frame}

\begin{frame}{Are decontamination approaches different?}
\phantomsection\label{are-decontamination-approaches-different}
Alpha Diversities

\begin{columns}[T]
\begin{column}{0.33\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap3/Chap3_Alpha_Obs.png}}
\end{column}

\begin{column}{0.33\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap3/Chap3_Alpha_Shannon.png}}
\end{column}

\begin{column}{0.33\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap3/Chap3_Alpha_InvS.png}}
\end{column}
\end{columns}
\end{frame}

\begin{frame}{Beta Diversity}
\phantomsection\label{beta-diversity-1}
\begin{columns}[T]
\begin{column}{0.33\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap3/Chap3_Beta_Raw_decontam.png}}
\end{column}

\begin{column}{0.33\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap3/Chap3_Beta_Raw_restrictuve.png}}
\end{column}

\begin{column}{0.33\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap3/Chap3_Beta_Raw_SCRuB.png}}
\end{column}
\end{columns}

\begin{columns}[T]
\begin{column}{0.33\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap3/Chap3_Beta_Raw_Nejman.png}}
\end{column}

\begin{column}{0.33\linewidth}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap3/Chap3_Beta_all.png}}
\end{column}
\end{columns}
\end{frame}

\begin{frame}{Bacterial Profile - Barplot}
\phantomsection\label{bacterial-profile---barplot-1}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap3/Chap3_barplot.png}}
\end{frame}

\begin{frame}{Bacterial Profile - Heatmap}
\phantomsection\label{bacterial-profile---heatmap-1}
\pandocbounded{\includegraphics[keepaspectratio]{img/Chap3/Chap3_heatmap.png}}
\end{frame}




\end{document}
