<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Linh Dang">

<title>Comprehensive Assessment</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="ComprehensiveAssessment_files/libs/clipboard/clipboard.min.js"></script>
<script src="ComprehensiveAssessment_files/libs/quarto-html/quarto.js"></script>
<script src="ComprehensiveAssessment_files/libs/quarto-html/popper.min.js"></script>
<script src="ComprehensiveAssessment_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="ComprehensiveAssessment_files/libs/quarto-html/anchor.min.js"></script>
<link href="ComprehensiveAssessment_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="ComprehensiveAssessment_files/libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="ComprehensiveAssessment_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="ComprehensiveAssessment_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="ComprehensiveAssessment_files/libs/bootstrap/bootstrap-34af8a4e30fcaa834c5f714e2bbc225f.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#comprehensive-assessment-of-decontamination-methods" id="toc-comprehensive-assessment-of-decontamination-methods" class="nav-link active" data-scroll-target="#comprehensive-assessment-of-decontamination-methods">Comprehensive Assessment of Decontamination Methods</a></li>
  <li><a href="#assessment-of-batch-effect-removal" id="toc-assessment-of-batch-effect-removal" class="nav-link" data-scroll-target="#assessment-of-batch-effect-removal">1. Assessment of batch effect removal</a>
  <ul class="collapse">
  <li><a href="#restrictive" id="toc-restrictive" class="nav-link" data-scroll-target="#restrictive">Restrictive</a></li>
  <li><a href="#decontam" id="toc-decontam" class="nav-link" data-scroll-target="#decontam">Decontam</a></li>
  <li><a href="#scrub" id="toc-scrub" class="nav-link" data-scroll-target="#scrub">SCRuB</a></li>
  <li><a href="#nejman" id="toc-nejman" class="nav-link" data-scroll-target="#nejman">Nejman</a></li>
  <li><a href="#overall-results" id="toc-overall-results" class="nav-link" data-scroll-target="#overall-results">Overall Results</a></li>
  </ul></li>
  <li><a href="#divergence-from-negative-controls" id="toc-divergence-from-negative-controls" class="nav-link" data-scroll-target="#divergence-from-negative-controls">2. Divergence from Negative Controls</a>
  <ul class="collapse">
  <li><a href="#raw" id="toc-raw" class="nav-link" data-scroll-target="#raw">Raw</a></li>
  <li><a href="#restrictive-1" id="toc-restrictive-1" class="nav-link" data-scroll-target="#restrictive-1">Restrictive</a></li>
  <li><a href="#decontam-1" id="toc-decontam-1" class="nav-link" data-scroll-target="#decontam-1">Decontam</a></li>
  <li><a href="#scrub-1" id="toc-scrub-1" class="nav-link" data-scroll-target="#scrub-1">SCRuB</a></li>
  <li><a href="#nejman-1" id="toc-nejman-1" class="nav-link" data-scroll-target="#nejman-1">Nejman</a></li>
  </ul></li>
  <li><a href="#biologically-meaningful-of-consensus-taxa" id="toc-biologically-meaningful-of-consensus-taxa" class="nav-link" data-scroll-target="#biologically-meaningful-of-consensus-taxa">3. Biologically meaningful of Consensus taxa</a></li>
  <li><a href="#overlap-with-known-true-taxa-contaminants" id="toc-overlap-with-known-true-taxa-contaminants" class="nav-link" data-scroll-target="#overlap-with-known-true-taxa-contaminants">4. Overlap with known true taxa / contaminants</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Comprehensive Assessment</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Linh Dang </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(microViz)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>microViz version 0.12.7 - Copyright (C) 2021-2025 David Barnett
! Website: https://david-barnett.github.io/microViz
✔ Useful?  For citation details, run: `citation("microViz")`
✖ Silence? `suppressPackageStartupMessages(library(microViz))`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(microbiome)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggplot2

microbiome R package (microbiome.github.com)
    


 Copyright (C) 2011-2022 Leo Lahti, 
    Sudarshan Shetty et al. &lt;microbiome.github.io&gt;


Attaching package: 'microbiome'

The following object is masked from 'package:ggplot2':

    alpha

The following object is masked from 'package:base':

    transform</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: permute

Attaching package: 'vegan'

The following object is masked from 'package:microbiome':

    diversity</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Wrench)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.1     ✔ stringr   1.5.2
✔ lubridate 1.9.4     ✔ tibble    3.3.0
✔ purrr     1.1.0     ✔ tidyr     1.3.1
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ microbiome::alpha() masks ggplot2::alpha()
✖ dplyr::filter()     masks stats::filter()
✖ dplyr::lag()        masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../../add_code/Functions.R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'rstatix'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:microViz':

    cor_test</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ggpubr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:microViz':

    stat_chull</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>df_additionalInfo <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">"../../meta/Mice_meta.xlsx"</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>pseq_raw <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../../data/Chap3/pseq_Proj5_postFilter_v04.rds"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ps_filter</span>(ffpe.bulk <span class="sc">==</span> <span class="st">"bulk"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ps_filter</span>(true.control <span class="sc">==</span> <span class="st">"true"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">append_AN_NR</span>(<span class="at">df_additional =</span> df_additionalInfo) <span class="sc">%&gt;%</span> </span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">WrenchWrapper</span>(<span class="at">grp =</span> <span class="st">"Sex"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Partial NA coefficients for 146 probe(s)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>pseq_restrictive <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../../data/Chap3/pseq_bulk_restrictive.rds"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">append_AN_NR</span>(df_additionalInfo) <span class="sc">%&gt;%</span> </span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">WrenchWrapper</span>(<span class="at">grp =</span> <span class="st">"Sex"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Partial NA coefficients for 73 probe(s)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>pseq_decontam <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../../data/Chap3/pseq_bulk_decontam_p0.5.rds"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">append_AN_NR</span>(df_additionalInfo) <span class="sc">%&gt;%</span> </span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">WrenchWrapper</span>(<span class="at">grp =</span> <span class="st">"Sex"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Partial NA coefficients for 97 probe(s)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>pseq_SCRuB <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../../data/Chap3/pseq_bulk_SCRuB.rds"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">append_AN_NR</span>(df_additionalInfo) <span class="sc">%&gt;%</span> </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">WrenchWrapper</span>(<span class="at">grp =</span> <span class="st">"Sex"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Partial NA coefficients for 300 probe(s)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>pseq_Nj <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../../data/Chap3/pseq_bulk_Fisher_v02.rds"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">append_AN_NR</span>(df_additionalInfo) <span class="sc">%&gt;%</span> </span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">WrenchWrapper</span>(<span class="at">grp =</span> <span class="st">"Sex"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</code></pre>
</div>
</div>
<section id="comprehensive-assessment-of-decontamination-methods" class="level2">
<h2 class="anchored" data-anchor-id="comprehensive-assessment-of-decontamination-methods">Comprehensive Assessment of Decontamination Methods</h2>
<p>Due to lacking of ground truth or mock community. Here are the alternative assess approaches we could use:</p>
<ul>
<li><p>Assessment of batch effect removal</p></li>
<li><p>Divergence from Negative Controls</p></li>
<li><p>Biologically meaningful of Consensus taxa</p></li>
<li><p>Overlap with known true taxa / contaminants.</p></li>
</ul>
</section>
<section id="assessment-of-batch-effect-removal" class="level2">
<h2 class="anchored" data-anchor-id="assessment-of-batch-effect-removal">1. Assessment of batch effect removal</h2>
<ul>
<li><p><strong>Main idea</strong>: A good decontamination method should reduce technical variation. In this case is the technician.</p></li>
<li><p><strong>Approach:</strong> Perform Ordination (PCoA) and PERMANOVA (adonis2).</p></li>
<li><p><strong>Metric:</strong> Check if the variance explained (R2) by “Sequencing Run” or “Extraction Batch” decreases after decontamination.</p></li>
<li><p><strong>Logic:</strong> Real tumor microbiomes shouldn’t cluster by which technician sequenced.</p></li>
</ul>
<p>Write a wrapper</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'gridExtra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    combine</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="do">### simple ord plot and permanova</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>assess_batch_effect <span class="ot">&lt;-</span> <span class="cf">function</span>(ps, <span class="at">method =</span> <span class="st">"PCoA"</span>, <span class="at">distance =</span> <span class="st">"bray"</span>, <span class="at">batch =</span> <span class="st">"person"</span>, <span class="at">gg_title =</span> <span class="st">"raw"</span>){</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  ord <span class="ot">&lt;-</span> <span class="fu">ordinate</span>(ps, <span class="at">method =</span> method, <span class="at">distance =</span> distance)</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ord_clean &lt;- ordinate(pseq_clean, method = "PCoA", distance = "bray")</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">plot_ordination</span>(ps, ord, <span class="at">color =</span> batch) <span class="sc">+</span> </span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>(<span class="at">type =</span> <span class="st">"norm"</span>) <span class="sc">+</span> <span class="co"># Adds confidence ellipses per batch</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(gg_title) <span class="sc">+</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>  <span class="do">## calc distance </span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>  dis <span class="ot">&lt;-</span> <span class="fu">distance</span>(ps, <span class="at">method =</span> distance)</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dist_clean &lt;- distance(pseq_clean, method = "bray")</span></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>  df_meta <span class="ot">&lt;-</span> <span class="fu">meta</span>(ps)</span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set.seed</span>(<span class="dv">210488</span>)</span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>  permanova_raw <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(<span class="fu">as.formula</span>(<span class="fu">paste0</span>(<span class="st">"dis ~ "</span>, batch)), <span class="at">data =</span> df_meta, <span class="at">permutations =</span> <span class="dv">9999</span>)</span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">=</span> <span class="fu">list</span>(<span class="at">plt =</span> p1, <span class="at">permanova =</span> permanova_raw)</span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="restrictive" class="level3">
<h3 class="anchored" data-anchor-id="restrictive">Restrictive</h3>
<section id="assessment-by-visualization" class="level4">
<h4 class="anchored" data-anchor-id="assessment-by-visualization">Assessment by visualization</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>res_raw <span class="ot">&lt;-</span> <span class="fu">assess_batch_effect</span>(pseq_raw, <span class="at">batch =</span> <span class="st">"person"</span>, <span class="at">gg_title =</span> <span class="st">"Raw - Before Decontamination"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `aes_string()` was deprecated in ggplot2 3.0.0.
ℹ Please use tidy evaluation idioms with `aes()`.
ℹ See also `vignette("ggplot2-in-packages")` for more information.
ℹ The deprecated feature was likely used in the phyloseq package.
  Please report the issue at &lt;https://github.com/joey711/phyloseq/issues&gt;.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>res_restrictive <span class="ot">&lt;-</span> <span class="fu">assess_batch_effect</span>(pseq_restrictive, <span class="at">batch =</span> <span class="st">"person"</span>, <span class="at">gg_title =</span> <span class="st">"After Decon. - Restrictive"</span>)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="co"># 4. Display side-by-side</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.arrange</span>(res_raw<span class="sc">$</span>plt, res_restrictive<span class="sc">$</span>plt, <span class="at">ncol =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="ComprehensiveAssessment_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="statistical-quantification-by-permanova" class="level4">
<h4 class="anchored" data-anchor-id="statistical-quantification-by-permanova">Statistical Quantification by PERMANOVA</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># --- PRINT RESULTS ---</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--- RAW DATA BATCH EFFECT ---"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "--- RAW DATA BATCH EFFECT ---"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(res_raw<span class="sc">$</span>permanova)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 9999

adonis2(formula = as.formula(paste0("dis ~ ", batch)), data = df_meta, permutations = 9999)
         Df SumOfSqs      R2      F Pr(&gt;F)
Model     1   0.4126 0.07527 1.3024 0.1776
Residual 16   5.0688 0.92473              
Total    17   5.4814 1.00000              </code></pre>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--- CLEAN DATA BATCH EFFECT ---"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "--- CLEAN DATA BATCH EFFECT ---"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(res_restrictive<span class="sc">$</span>permanova)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 9999

adonis2(formula = as.formula(paste0("dis ~ ", batch)), data = df_meta, permutations = 9999)
         Df SumOfSqs      R2      F Pr(&gt;F)  
Model     1   0.6762 0.09738 1.7261 0.0195 *
Residual 16   6.2679 0.90262                
Total    17   6.9441 1.00000                
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
</section>
</section>
<section id="decontam" class="level3">
<h3 class="anchored" data-anchor-id="decontam">Decontam</h3>
<section id="assessment-by-visualization-1" class="level4">
<h4 class="anchored" data-anchor-id="assessment-by-visualization-1">Assessment by visualization</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>res_decontam <span class="ot">&lt;-</span> <span class="fu">assess_batch_effect</span>(pseq_decontam, <span class="at">gg_title =</span> <span class="st">"After Decon. - Decontam"</span>)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.arrange</span>(res_raw<span class="sc">$</span>plt, res_decontam<span class="sc">$</span>plt, <span class="at">ncol =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="ComprehensiveAssessment_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="statistical-quantification-by-permanova-1" class="level4">
<h4 class="anchored" data-anchor-id="statistical-quantification-by-permanova-1">Statistical Quantification by PERMANOVA</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># --- PRINT RESULTS ---</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--- RAW DATA BATCH EFFECT ---"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "--- RAW DATA BATCH EFFECT ---"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(res_raw<span class="sc">$</span>permanova)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 9999

adonis2(formula = as.formula(paste0("dis ~ ", batch)), data = df_meta, permutations = 9999)
         Df SumOfSqs      R2      F Pr(&gt;F)
Model     1   0.4126 0.07527 1.3024 0.1776
Residual 16   5.0688 0.92473              
Total    17   5.4814 1.00000              </code></pre>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--- CLEAN DATA BATCH EFFECT ---"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "--- CLEAN DATA BATCH EFFECT ---"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(res_decontam<span class="sc">$</span>permanova)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 9999

adonis2(formula = as.formula(paste0("dis ~ ", batch)), data = df_meta, permutations = 9999)
         Df SumOfSqs      R2      F Pr(&gt;F)  
Model     1   0.6312 0.08875 1.5583 0.0352 *
Residual 16   6.4813 0.91125                
Total    17   7.1126 1.00000                
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
</section>
</section>
<section id="scrub" class="level3">
<h3 class="anchored" data-anchor-id="scrub">SCRuB</h3>
<section id="assessment-by-visualization-2" class="level4">
<h4 class="anchored" data-anchor-id="assessment-by-visualization-2">Assessment by visualization</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>res_SCRuB <span class="ot">&lt;-</span> <span class="fu">assess_batch_effect</span>(pseq_SCRuB, <span class="at">gg_title =</span> <span class="st">"After Decon. - SCRuB"</span>)</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.arrange</span>(res_raw<span class="sc">$</span>plt, res_SCRuB<span class="sc">$</span>plt, <span class="at">ncol =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="ComprehensiveAssessment_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="statistical-quantification-by-permanova-2" class="level4">
<h4 class="anchored" data-anchor-id="statistical-quantification-by-permanova-2">Statistical Quantification by PERMANOVA</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># --- PRINT RESULTS ---</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--- RAW DATA BATCH EFFECT ---"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "--- RAW DATA BATCH EFFECT ---"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(res_raw<span class="sc">$</span>permanova)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 9999

adonis2(formula = as.formula(paste0("dis ~ ", batch)), data = df_meta, permutations = 9999)
         Df SumOfSqs      R2      F Pr(&gt;F)
Model     1   0.4126 0.07527 1.3024 0.1776
Residual 16   5.0688 0.92473              
Total    17   5.4814 1.00000              </code></pre>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--- CLEAN DATA BATCH EFFECT ---"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "--- CLEAN DATA BATCH EFFECT ---"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(res_SCRuB<span class="sc">$</span>permanova)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 9999

adonis2(formula = as.formula(paste0("dis ~ ", batch)), data = df_meta, permutations = 9999)
         Df SumOfSqs      R2      F Pr(&gt;F)  
Model     1   0.6650 0.09197 1.6205 0.0291 *
Residual 16   6.5661 0.90803                
Total    17   7.2311 1.00000                
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
</section>
</section>
<section id="nejman" class="level3">
<h3 class="anchored" data-anchor-id="nejman">Nejman</h3>
<section id="assessment-by-visualization-3" class="level4">
<h4 class="anchored" data-anchor-id="assessment-by-visualization-3">Assessment by visualization</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>res_Nj <span class="ot">&lt;-</span> <span class="fu">assess_batch_effect</span>(pseq_Nj, <span class="at">gg_title =</span> <span class="st">"After Decon. - Nejman"</span>)</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.arrange</span>(res_raw<span class="sc">$</span>plt, res_Nj<span class="sc">$</span>plt, <span class="at">ncol =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="ComprehensiveAssessment_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="statistical-quantification-by-permanova-3" class="level4">
<h4 class="anchored" data-anchor-id="statistical-quantification-by-permanova-3">Statistical Quantification by PERMANOVA</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># --- PRINT RESULTS ---</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--- RAW DATA BATCH EFFECT ---"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "--- RAW DATA BATCH EFFECT ---"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(res_raw<span class="sc">$</span>permanova)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 9999

adonis2(formula = as.formula(paste0("dis ~ ", batch)), data = df_meta, permutations = 9999)
         Df SumOfSqs      R2      F Pr(&gt;F)
Model     1   0.4126 0.07527 1.3024 0.1776
Residual 16   5.0688 0.92473              
Total    17   5.4814 1.00000              </code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"--- CLEAN DATA BATCH EFFECT ---"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "--- CLEAN DATA BATCH EFFECT ---"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(res_Nj<span class="sc">$</span>permanova)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 9999

adonis2(formula = as.formula(paste0("dis ~ ", batch)), data = df_meta, permutations = 9999)
         Df SumOfSqs      R2      F Pr(&gt;F)  
Model     1   0.6047 0.09167 1.6147 0.0486 *
Residual 16   5.9922 0.90833                
Total    17   6.5970 1.00000                
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
</section>
</section>
<section id="overall-results" class="level3">
<h3 class="anchored" data-anchor-id="overall-results">Overall Results</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>df_overall_R2_pval <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">Method =</span> <span class="fu">character</span>(<span class="dv">0</span>),</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">R2 =</span> <span class="fu">double</span>(<span class="dv">0</span>),</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">p_value =</span> <span class="fu">double</span>(<span class="dv">0</span>))</span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>df_overall_R2_pval <span class="ot">&lt;-</span> df_overall_R2_pval <span class="sc">%&gt;%</span> </span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_row</span>(<span class="at">Method =</span> <span class="st">"Raw"</span>, <span class="at">R2 =</span> res_raw<span class="sc">$</span>permanova<span class="sc">$</span>R2[<span class="dv">1</span>], <span class="at">p_value =</span> res_raw<span class="sc">$</span>permanova<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_row</span>(<span class="at">Method =</span> <span class="st">"Restrictive"</span>, <span class="at">R2 =</span> res_restrictive<span class="sc">$</span>permanova<span class="sc">$</span>R2[<span class="dv">1</span>], <span class="at">p_value =</span> res_restrictive<span class="sc">$</span>permanova<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_row</span>(<span class="at">Method =</span> <span class="st">"decontam"</span>, <span class="at">R2 =</span> res_decontam<span class="sc">$</span>permanova<span class="sc">$</span>R2[<span class="dv">1</span>], <span class="at">p_value =</span> res_decontam<span class="sc">$</span>permanova<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_row</span>(<span class="at">Method =</span> <span class="st">"SCRuB"</span>, <span class="at">R2 =</span> res_SCRuB<span class="sc">$</span>permanova<span class="sc">$</span>R2[<span class="dv">1</span>], <span class="at">p_value =</span> res_SCRuB<span class="sc">$</span>permanova<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_row</span>(<span class="at">Method =</span> <span class="st">"Nejman"</span>, <span class="at">R2 =</span> res_Nj<span class="sc">$</span>permanova<span class="sc">$</span>R2[<span class="dv">1</span>], <span class="at">p_value =</span> res_Nj<span class="sc">$</span>permanova<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>])</span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(df_overall_R2_pval)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 5 × 3
  Method          R2 p_value
  &lt;chr&gt;        &lt;dbl&gt;   &lt;dbl&gt;
1 Raw         0.0753  0.178 
2 Restrictive 0.0974  0.0195
3 decontam    0.0887  0.0352
4 SCRuB       0.0920  0.0291
5 Nejman      0.0917  0.0486</code></pre>
</div>
</div>
<p>Conclusion: For this dataset, this is not the method we like to choose.</p>
</section>
</section>
<section id="divergence-from-negative-controls" class="level2">
<h2 class="anchored" data-anchor-id="divergence-from-negative-controls">2. Divergence from Negative Controls</h2>
<ul>
<li><p><strong>Approach:</strong> perform Inter-group dissimilarity and Wilcox paired test, following by visualization.</p></li>
<li><p><strong>Metric:</strong> Calculate the Jensen-Shannon Divergence (or Bray-Curtis) between samples and the pool of negative controls.</p></li>
<li><p><strong>Logic:</strong> the decontamination method should maximize the dissimilarity between true PDAC samples and NCTs.</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>pseq_nct <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../../data/Chap3/pseq_Proj5_postFilter_v04.rds"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ps_filter</span>(ffpe.bulk <span class="sc">==</span> <span class="st">"bulk"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ps_filter</span>(true.control <span class="sc">!=</span> <span class="st">"true"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Because we could not apply wrench normalization for NCTs, thus we have to use rarefaction</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>s_size <span class="ot">&lt;-</span> <span class="dv">5000</span></span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>pseq_raw_both <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../../data/Chap3/pseq_Proj5_postFilter_v04.rds"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ps_filter</span>(ffpe.bulk <span class="sc">==</span> <span class="st">"bulk"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rarefy_even_depth</span>(<span class="at">sample.size =</span> s_size)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>You set `rngseed` to FALSE. Make sure you've set &amp; recorded
 the random seed of your session for reproducibility.
See `?set.seed`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>4 samples removedbecause they contained fewer reads than `sample.size`.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Up to first five removed samples are: </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode112022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode232022_03_09_16s_kpc_tum-panc_invitro_ntc_fast5_barcode102024_03_27_AS_Isa_16s_NCT_fast5_barcode05</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>125OTUs were removed because they are no longer 
present in any sample after random subsampling</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>pseq_restrictive_both <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../../data/Chap3/pseq_bulk_restrictive.rds"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge_phyloseq</span>(pseq_nct) <span class="sc">%&gt;%</span> </span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rarefy_even_depth</span>(<span class="at">sample.size =</span> s_size)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>You set `rngseed` to FALSE. Make sure you've set &amp; recorded
 the random seed of your session for reproducibility.
See `?set.seed`

...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>7 samples removedbecause they contained fewer reads than `sample.size`.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Up to first five removed samples are: </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode052022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode112022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode192022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode202022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode23</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>124OTUs were removed because they are no longer 
present in any sample after random subsampling</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>pseq_decontam_both <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../../data/Chap3/pseq_bulk_decontam_p0.5.rds"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge_phyloseq</span>(pseq_nct) <span class="sc">%&gt;%</span> </span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rarefy_even_depth</span>(<span class="at">sample.size =</span> s_size)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>You set `rngseed` to FALSE. Make sure you've set &amp; recorded
 the random seed of your session for reproducibility.
See `?set.seed`

...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>7 samples removedbecause they contained fewer reads than `sample.size`.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Up to first five removed samples are: </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode052022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode112022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode192022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode202022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode23</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>98OTUs were removed because they are no longer 
present in any sample after random subsampling</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>pseq_SCRuB_both <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../../data/Chap3/pseq_bulk_SCRuB.rds"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge_phyloseq</span>(pseq_nct) <span class="sc">%&gt;%</span> </span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rarefy_even_depth</span>(<span class="at">sample.size =</span> s_size)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>You set `rngseed` to FALSE. Make sure you've set &amp; recorded
 the random seed of your session for reproducibility.
See `?set.seed`

...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>6 samples removedbecause they contained fewer reads than `sample.size`.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Up to first five removed samples are: </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode052022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode112022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode202022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode232022_03_09_16s_kpc_tum-panc_invitro_ntc_fast5_barcode10</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>83OTUs were removed because they are no longer 
present in any sample after random subsampling</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>pseq_Nj_both <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../../data/Chap3/pseq_bulk_Fisher_v02.rds"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge_phyloseq</span>(pseq_nct) <span class="sc">%&gt;%</span> </span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rarefy_even_depth</span>(<span class="at">sample.size =</span> s_size)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>You set `rngseed` to FALSE. Make sure you've set &amp; recorded
 the random seed of your session for reproducibility.
See `?set.seed`

...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>8 samples removedbecause they contained fewer reads than `sample.size`.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Up to first five removed samples are: </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>2022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode052022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode112022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode192022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode202022_02_23_16s_kpc_tum-panc_invitro_fast5_barcode23</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>95OTUs were removed because they are no longer 
present in any sample after random subsampling</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>...</code></pre>
</div>
</div>
<p>Write a wrapper</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a>get_dist_to_controls <span class="ot">&lt;-</span> <span class="cf">function</span>(ps, group_col, <span class="at">true_label =</span> <span class="st">"true"</span>, <span class="at">control_label =</span> <span class="st">"NCT"</span>, <span class="at">metric =</span> <span class="st">"bray"</span>){</span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">tolower</span>(metric) <span class="sc">==</span> <span class="st">"jsd"</span>) {</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a>    ps <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(ps, <span class="cf">function</span>(x) x <span class="sc">/</span> <span class="fu">sum</span>(x))</span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>  dist_mat <span class="ot">&lt;-</span> ps <span class="sc">%&gt;%</span> </span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distance</span>(<span class="at">method =</span> metric) <span class="sc">%&gt;%</span> </span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.matrix</span>()</span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a>  <span class="co">#print(dist_mat)</span></span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a>  meta <span class="ot">&lt;-</span> <span class="fu">meta</span>(ps)</span>
<span id="cb114-10"><a href="#cb114-10" aria-hidden="true" tabindex="-1"></a>  true_ids <span class="ot">&lt;-</span> <span class="fu">rownames</span>(meta)[meta[[group_col]] <span class="sc">==</span> true_label]</span>
<span id="cb114-11"><a href="#cb114-11" aria-hidden="true" tabindex="-1"></a>  ctrl_ids <span class="ot">&lt;-</span> <span class="fu">rownames</span>(meta)[meta[[group_col]] <span class="sc">==</span> control_label]</span>
<span id="cb114-12"><a href="#cb114-12" aria-hidden="true" tabindex="-1"></a>  <span class="co">#print(ctrl_ids)</span></span>
<span id="cb114-13"><a href="#cb114-13" aria-hidden="true" tabindex="-1"></a>  sub_mat <span class="ot">&lt;-</span> dist_mat[true_ids, ctrl_ids, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb114-14"><a href="#cb114-14" aria-hidden="true" tabindex="-1"></a>  <span class="co">#print(sub_mat)</span></span>
<span id="cb114-15"><a href="#cb114-15" aria-hidden="true" tabindex="-1"></a>  mean_dists <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(sub_mat, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb114-16"><a href="#cb114-16" aria-hidden="true" tabindex="-1"></a>  <span class="co">#print(mean_dists)</span></span>
<span id="cb114-17"><a href="#cb114-17" aria-hidden="true" tabindex="-1"></a>  median_dists <span class="ot">&lt;-</span> <span class="fu">apply</span>(sub_mat, <span class="dv">1</span>, median, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb114-18"><a href="#cb114-18" aria-hidden="true" tabindex="-1"></a>  <span class="co">#print(median_dists)</span></span>
<span id="cb114-19"><a href="#cb114-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">SampleID =</span> <span class="fu">names</span>(mean_dists), <span class="at">MeanDist =</span> mean_dists, <span class="at">MedianDist =</span> median_dists))</span>
<span id="cb114-20"><a href="#cb114-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb114-21"><a href="#cb114-21" aria-hidden="true" tabindex="-1"></a><span class="do">### Wrapper of statistics and ggplots</span></span>
<span id="cb114-22"><a href="#cb114-22" aria-hidden="true" tabindex="-1"></a>wrapper_dist_to_controls <span class="ot">&lt;-</span> <span class="cf">function</span>(ps_raw, ps_clean, group_col, <span class="at">true_label =</span> <span class="st">"true"</span>, <span class="at">control_label =</span> <span class="st">"NCT"</span>, <span class="at">metric =</span> <span class="st">"bray"</span>, <span class="at">decontam_name =</span> <span class="st">"None"</span>){</span>
<span id="cb114-23"><a href="#cb114-23" aria-hidden="true" tabindex="-1"></a>  df_raw <span class="ot">&lt;-</span> <span class="fu">get_dist_to_controls</span>(ps_raw, group_col, true_label, control_label, metric)</span>
<span id="cb114-24"><a href="#cb114-24" aria-hidden="true" tabindex="-1"></a>  df_clean <span class="ot">&lt;-</span> <span class="fu">get_dist_to_controls</span>(ps_clean, group_col, true_label, control_label, metric)</span>
<span id="cb114-25"><a href="#cb114-25" aria-hidden="true" tabindex="-1"></a>  df_raw <span class="ot">&lt;-</span> df_raw <span class="sc">%&gt;%</span> <span class="fu">select</span>(SampleID, <span class="at">MeanDist_raw =</span> MeanDist)</span>
<span id="cb114-26"><a href="#cb114-26" aria-hidden="true" tabindex="-1"></a>  df_clean <span class="ot">&lt;-</span> df_clean <span class="sc">%&gt;%</span> <span class="fu">select</span>(SampleID, <span class="at">MeanDist_clean =</span> MeanDist)</span>
<span id="cb114-27"><a href="#cb114-27" aria-hidden="true" tabindex="-1"></a>  <span class="co">#print()</span></span>
<span id="cb114-28"><a href="#cb114-28" aria-hidden="true" tabindex="-1"></a>  df_stats <span class="ot">&lt;-</span> <span class="fu">merge</span>(df_raw, df_clean, <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb114-29"><a href="#cb114-29" aria-hidden="true" tabindex="-1"></a>  test_results <span class="ot">&lt;-</span> <span class="fu">wilcox.test</span>(df_stats<span class="sc">$</span>MeanDist_clean, df_stats<span class="sc">$</span>MeanDist_raw,</span>
<span id="cb114-30"><a href="#cb114-30" aria-hidden="true" tabindex="-1"></a>                            <span class="at">paired =</span> <span class="cn">TRUE</span>, <span class="at">alternative =</span> <span class="st">"greater"</span>)</span>
<span id="cb114-31"><a href="#cb114-31" aria-hidden="true" tabindex="-1"></a>  df_long <span class="ot">&lt;-</span> <span class="fu">melt</span>(df_stats, <span class="at">id.vars=</span><span class="st">"SampleID"</span>, <span class="at">measure.vars=</span><span class="fu">c</span>(<span class="st">"MeanDist_raw"</span>, <span class="st">"MeanDist_clean"</span>))</span>
<span id="cb114-32"><a href="#cb114-32" aria-hidden="true" tabindex="-1"></a>  plt <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_long, <span class="fu">aes</span>(<span class="at">x=</span>variable, <span class="at">y=</span>value, <span class="at">fill=</span>variable)) <span class="sc">+</span></span>
<span id="cb114-33"><a href="#cb114-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">alpha=</span><span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb114-34"><a href="#cb114-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">group=</span>SampleID), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span> <span class="co"># Adds the dots</span></span>
<span id="cb114-35"><a href="#cb114-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group=</span>SampleID), <span class="at">alpha=</span><span class="fl">0.2</span>) <span class="sc">+</span>  <span class="co"># Connects the dots (shows paired change)</span></span>
<span id="cb114-36"><a href="#cb114-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb114-37"><a href="#cb114-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Divergence from Negative Controls"</span>,</span>
<span id="cb114-38"><a href="#cb114-38" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="fu">paste</span>(<span class="st">"P-value:"</span>, <span class="fu">format.pval</span>(test_results<span class="sc">$</span>p.value, <span class="at">digits=</span><span class="dv">4</span>)),</span>
<span id="cb114-39"><a href="#cb114-39" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Mean JSD to Negative Controls"</span>,</span>
<span id="cb114-40"><a href="#cb114-40" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Method Stage"</span>) <span class="sc">+</span></span>
<span id="cb114-41"><a href="#cb114-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Raw"</span>, decontam_name))</span>
<span id="cb114-42"><a href="#cb114-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">test_results =</span> test_results, <span class="at">plt =</span> plt))</span>
<span id="cb114-43"><a href="#cb114-43" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="raw" class="level3">
<h3 class="anchored" data-anchor-id="raw">Raw</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="do">## calc distnace of each sample to negative control group</span></span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a>df_raw <span class="ot">&lt;-</span> <span class="fu">get_dist_to_controls</span>(pseq_raw_both, <span class="at">group_col =</span> <span class="st">"true.control"</span>,</span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a>                               <span class="at">true_label =</span> <span class="st">"true"</span>, <span class="at">control_label =</span> <span class="st">"NCT"</span>,</span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a>                               <span class="at">metric =</span> <span class="st">"jsd"</span>)</span>
<span id="cb115-5"><a href="#cb115-5" aria-hidden="true" tabindex="-1"></a>df_raw <span class="ot">&lt;-</span> df_raw <span class="sc">%&gt;%</span> </span>
<span id="cb115-6"><a href="#cb115-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">MeanDist_raw =</span> MeanDist, <span class="at">MedianDist_raw =</span> MedianDist)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="restrictive-1" class="level3">
<h3 class="anchored" data-anchor-id="restrictive-1">Restrictive</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">wrapper_dist_to_controls</span>(pseq_raw_both, pseq_restrictive_both,</span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">group_col =</span> <span class="st">"true.control"</span>, </span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">true_label =</span> <span class="st">"true"</span>, <span class="at">control_label =</span> <span class="st">"NCT"</span>,</span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">metric =</span> <span class="st">"jsd"</span>, <span class="at">decontam_name =</span> <span class="st">"Restrictive"</span>)</span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(res<span class="sc">$</span>test_results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Wilcoxon signed rank exact test

data:  df_stats$MeanDist_clean and df_stats$MeanDist_raw
V = 91, p-value = 0.0001221
alternative hypothesis: true location shift is greater than 0</code></pre>
</div>
</div>
<p>GGplot</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>plt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="ComprehensiveAssessment_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="decontam-1" class="level3">
<h3 class="anchored" data-anchor-id="decontam-1">Decontam</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">wrapper_dist_to_controls</span>(pseq_raw_both, pseq_decontam_both,</span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">group_col =</span> <span class="st">"true.control"</span>, </span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">true_label =</span> <span class="st">"true"</span>, <span class="at">control_label =</span> <span class="st">"NCT"</span>,</span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">metric =</span> <span class="st">"jsd"</span>, <span class="at">decontam_name =</span> <span class="st">"decontam"</span>)</span>
<span id="cb119-5"><a href="#cb119-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(res<span class="sc">$</span>test_results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Wilcoxon signed rank exact test

data:  df_stats$MeanDist_clean and df_stats$MeanDist_raw
V = 91, p-value = 0.0001221
alternative hypothesis: true location shift is greater than 0</code></pre>
</div>
</div>
<p>GGplot</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>plt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="ComprehensiveAssessment_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="scrub-1" class="level3">
<h3 class="anchored" data-anchor-id="scrub-1">SCRuB</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">wrapper_dist_to_controls</span>(pseq_raw_both, pseq_SCRuB_both,</span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">group_col =</span> <span class="st">"true.control"</span>, </span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">true_label =</span> <span class="st">"true"</span>, <span class="at">control_label =</span> <span class="st">"NCT"</span>,</span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">metric =</span> <span class="st">"jsd"</span>, <span class="at">decontam_name =</span> <span class="st">"SCRuB"</span>)</span>
<span id="cb122-5"><a href="#cb122-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(res<span class="sc">$</span>test_results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Wilcoxon signed rank exact test

data:  df_stats$MeanDist_clean and df_stats$MeanDist_raw
V = 105, p-value = 6.104e-05
alternative hypothesis: true location shift is greater than 0</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>plt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="ComprehensiveAssessment_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="nejman-1" class="level3">
<h3 class="anchored" data-anchor-id="nejman-1">Nejman</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">wrapper_dist_to_controls</span>(pseq_raw_both, pseq_Nj_both,</span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">group_col =</span> <span class="st">"true.control"</span>, </span>
<span id="cb125-3"><a href="#cb125-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">true_label =</span> <span class="st">"true"</span>, <span class="at">control_label =</span> <span class="st">"NCT"</span>,</span>
<span id="cb125-4"><a href="#cb125-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">metric =</span> <span class="st">"jsd"</span>, <span class="at">decontam_name =</span> <span class="st">"Nejman"</span>)</span>
<span id="cb125-5"><a href="#cb125-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(res<span class="sc">$</span>test_results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Wilcoxon signed rank exact test

data:  df_stats$MeanDist_clean and df_stats$MeanDist_raw
V = 78, p-value = 0.0002441
alternative hypothesis: true location shift is greater than 0</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>plt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="ComprehensiveAssessment_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Conclusion: all method performer well!</p>
</section>
</section>
<section id="biologically-meaningful-of-consensus-taxa" class="level2">
<h2 class="anchored" data-anchor-id="biologically-meaningful-of-consensus-taxa">3. Biologically meaningful of Consensus taxa</h2>
</section>
<section id="overlap-with-known-true-taxa-contaminants" class="level2">
<h2 class="anchored" data-anchor-id="overlap-with-known-true-taxa-contaminants">4. Overlap with known true taxa / contaminants</h2>
<ul>
<li><p><strong>Approach:</strong></p></li>
<li><p><strong>Metric:</strong></p></li>
<li><p><strong>Logic:</strong> a good decontamination methods should give results whose taxa is less overlap with known high prevalence contaminants in large set of NCTs.</p></li>
</ul>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>